persistence/sql/batch/create.go

```diff
@@ -211,23 +262,55 @@ func Create[T any](ctx context.Context, p *TracerConnection, models []*T) (err e
 	if err != nil {
 		return sqlcon.HandleError(err)
 	}
-	defer rows.Close()
```

Comment:
With some of the error handling we could now return early and the rows will not be closed. The documentation suggests that you are allowed to close the rows multiple times, so I'd keep the defer statement here just to ensure they are always closed.

> Close is idempotent and does not affect the result of `Rows.Err`. [from here](https://pkg.go.dev/database/sql#Rows.Close)

> Failing to read all rows or use rows.Close() reserves connections from the pool. [from here](http://go-database-sql.org/surprises.html)